octopusdeploy_framework/resource_azure_container_registry_test.go

```diff
@@ -133,3 +133,42 @@ func testAzureFeedCheckDestroy(s *terraform.State) error {
 
 	return nil
 }
+
+func TestAccOctopusDeployAzureFeedWithOIDC(t *testing.T) {
+	localName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	prefix := "octopusdeploy_azure_container_registry." + localName
+
+	oidcConfig := `
+resource "octopusdeploy_azure_container_registry" "%s" {
+  name         = "Azure OIDC Registry"
+  feed_uri     = "https://test-azure-oidc.azurecr.io"
+  registry_path = "test-oidc-registry"
+  oidc_authentication = {
+    client_id   = "00000000-0000-0000-0000-000000000000"
+    tenant_id   = "11111111-1111-1111-1111-111111111111"
+    audience    = "audience"
+    subject_keys = ["feed", "space"]
+  }
+}
+`
+	resource.Test(t, resource.TestCase{
+		CheckDestroy:             func(s *terraform.State) error { return testAzureFeedCheckDestroy(s) },
+		PreCheck:                 func() { TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: ProtoV6ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(oidcConfig, localName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(prefix, "name", "Azure OIDC Registry"),
+					resource.TestCheckResourceAttr(prefix, "feed_uri", "https://test-azure-oidc.azurecr.io"),
+					resource.TestCheckResourceAttr(prefix, "registry_path", "test-oidc-registry"),
+					resource.TestCheckResourceAttr(prefix, "oidc_authentication.client_id", "00000000-0000-0000-0000-000000000000"),
+					resource.TestCheckResourceAttr(prefix, "oidc_authentication.tenant_id", "11111111-1111-1111-1111-111111111111"),
+					resource.TestCheckResourceAttr(prefix, "oidc_authentication.audience", "audience"),
+					resource.TestCheckTypeSetElemAttr(prefix, "oidc_authentication.subject_keys.*", "feed"),
+					resource.TestCheckTypeSetElemAttr(prefix, "oidc_authentication.subject_keys.*", "space"),
+				),
+			},
```

Comment:
consider making a change to the config and running it in another TestStep object to ensure update works correctly.